lib/AST/Decl.cpp

```diff
@@ -6378,10 +6374,6 @@ Type ValueDecl::getFunctionBuilderType() const {
 }
 
 CustomAttr *ValueDecl::getAttachedFunctionBuilder() const {
-  // Fast path: most declarations (especially parameters, which is where
-  // this is hottest) do not have any custom attributes at all.
-  if (!getAttrs().hasAttribute<CustomAttr>()) return nullptr;
-
```

Comment:
Losing these hot paths seems abstractly unfortunate.  I never profiled them, of course, so who knows, but this does get called for every parameter of every direct call, at least when the argument is a closure.  Have you measured compile time/memory impact?

You're only inferring function builders for method bodies, I think, which should let us more selectively weaken the fast-path check.

Reply:
The request-evaluator needs to be fast enough so that we don't need to scatter hot paths like this everywhere, because they suppress incremental dependency generation.

Reply:
I feel like "needs to be fast enough" is doing a lot of work for something that inherently looks up and allocates a persistent hash-table entry.